src/components/CreatePostInput/index.js

```diff
@@ -14,34 +15,29 @@ import {joinTopicIntoTopicList} from '../../utils/array/ChunkArray';
 
 const CreatePostInput = ({
   allTaggedUser = [],
-  message = '',
+  message,
   topics = [],
   topicChats = [],
-  setAllTaggedUser = () => {},
-  setMessage = () => {},
-  setPositionKeyboard = () => {},
-  setTopics = () => {},
-  setTopicChats = () => {}
+  setAllTaggedUser,
+  setPositionKeyboard,
+  setTopics,
+  setTopicChats,
+  setMessage
 }) => {
   const [positionEndCursor, setPositionEndCursor] = React.useState(0);
   const [topicSearch, setTopicSearch] = React.useState([]);
   const [userTaggingSearch, setUserTaggingSearch] = React.useState([]);
   const [hashtagPosition, setHashtagPosition] = React.useState(0);
   const [positionTopicSearch, setPositionTopicSearch] = React.useState(0);
   const [lastTopicInCursor, setLastTopicInCursor] = React.useState('');
-  const [shouldUpdateHashtag, setShouldUpdateHashtag] = React.useState(0);
   const inputRef = React.useRef();
   const {formattedText, handleStateHashtag, handleStateMention, setHashtags, updateHashtag} =
     useHastagMention('');
 
   React.useEffect(() => {
-    setShouldUpdateHashtag(new Date().valueOf());
+    if (new Date().valueOf() > 0) updateHashtag(message, topics, setTopics, positionEndCursor);
   }, [message]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The refactor here simplifies the code by removing unnecessary state variables and effects. The removal of `shouldUpdateHashtag` and its associated `useEffect` makes the code cleaner and easier to understand. The direct call to `updateHashtag` in the `useEffect` dependent on `message` is more straightforward.
```diff
-  const [shouldUpdateHashtag, setShouldUpdateHashtag] = React.useState(0);
-  React.useEffect(() => {
-    setShouldUpdateHashtag(new Date().valueOf());
-  }, [message]);
-  React.useEffect(() => {
-    if (shouldUpdateHashtag > 0) updateHashtag(message, topics, setTopics, positionEndCursor);
-  }, [shouldUpdateHashtag]);
+  React.useEffect(() => {
+    if (new Date().valueOf() > 0) updateHashtag(message, topics, setTopics, positionEndCursor);
+  }, [message]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->